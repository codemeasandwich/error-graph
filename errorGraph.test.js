const errorGraph = require('./errorGraph')

const errors = [
`Error: request problem
 at Object.<anonymous> (/Users/user/Documents/www/file.js:4:12)
 at Module._compile (internal/modules/cjs/loader.js:774:30)
 at Object.Module._extensions..js (internal/modules/cjs/loader.js:785:10)
 at Module.load (internal/modules/cjs/loader.js:641:32)
 at Function.Module._load (internal/modules/cjs/loader.js:556:12)
 at Function.Module.runMain (internal/modules/cjs/loader.js:837:10)
 at internal/main/run_main_module.js:17:11`,
`Error: request problem
 at Object.<anonymous> (/Users/user/Documents/www/file.js:4:12)`,
`Error: request problem
 at Object.<anonymous> (/Users/user/Documents/www/file.js:44:3)`,
`Error: user login problem
 at Object.<anonymous> (/Users/user/Documents/www/foo.js:4:12)`,
`Error: user password problem
 at Object.<anonymous> (/Users/user/Documents/www/foo.js:4:12)`
]

test('errors as stack traces', () => {

  errors.forEach(stack => errorGraph(stack))
  console.log(errorGraph.dump())
  expect(errorGraph.dump()).toMatchSnapshot();
});
