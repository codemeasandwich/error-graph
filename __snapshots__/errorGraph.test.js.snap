// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors as stack traces 1`] = `
Object {
  "Error": Object {
    "/Users/user/Documents/www/file.js:44:3": Object {
      " request problem": Object {
        "__meta": Object {
          "\\"?\\"": 1,
        },
      },
      "__meta": Object {
        " request problem": 1,
        "__relationship": Array [],
      },
    },
    "/Users/user/Documents/www/file.js:4:12": Object {
      " request problem": Object {
        "__meta": Object {
          "\\"?\\"": 2,
        },
      },
      "__meta": Object {
        " request problem": 2,
        "__relationship": Array [],
      },
    },
    "/Users/user/Documents/www/foo.js:4:12": Object {
      " user login problem": Object {
        "__meta": Object {
          "\\"?\\"": 1,
        },
      },
      " user password problem": Object {
        "__meta": Object {
          "\\"?\\"": 1,
        },
      },
      "__meta": Object {
        " user login problem": 1,
        " user password problem": 1,
        "__relationship": Array [
          Array [
            0.6363636363636364,
            " user login problem",
            " user password problem",
          ],
        ],
      },
    },
    "__meta": Object {
      "/Users/user/Documents/www/file.js:44:3": 1,
      "/Users/user/Documents/www/file.js:4:12": 2,
      "/Users/user/Documents/www/foo.js:4:12": 2,
    },
  },
  "__meta": Object {
    "Error": 5,
    "__info": Object {},
  },
}
`;
